sdk/database/helper/connutil/sql.go

```diff
@@ -56,15 +56,22 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 	}
 
 	// Don't escape special characters for MySQL password
+	// Also don't escape special characters for the username and password if
+	// the disable_escaping parameter is set to true
+	disableEscaping := c.RawConfig["disable_escaping"].(bool)
```

Comment:
I wonder if we want to add `disable_escaping` to [SQLConnectionProducer](https://github.com/hashicorp/vault/blob/main/sdk/database/helper/connutil/sql.go#L22-L28) and let mapstructure do the [decoding](https://github.com/hashicorp/vault/blob/main/sdk/database/helper/connutil/sql.go#L49) for us? Right now, we're pulling the configuration value out of `RawConfig`, which may be okay. It appears that everything else is being decoded by mapstructure though.